Modix.Data.Test/Repositories/EmojiRepositoryTests.cs

```diff
@@ -0,0 +1,300 @@
+using Modix.Data.Repositories;
+using Modix.Data.Test.TestData;
+using Modix.Data.Test.TestData.Emoji;
+
+using NSubstitute;
+
+using NUnit.Framework;
+
+using Shouldly;
+
+namespace Modix.Data.Test.Repositories
+{
+    internal class EmojiRepositoryTests
+    {
+        #region Test Context
+
+        private static (ModixContext, EmojiRepository) BuildTestContext()
+        {
+            var modixContext = TestDataContextFactory.BuildTestDataContext(x =>
+            {
+                x.Emoji.AddRange(Emoji.Entities.Clone());
+            });
+
+            var uut = new EmojiRepository(modixContext);
+
+            return (modixContext, uut);
+        }
+
+        #endregion Test Context
+
+        #region Constructor() Tests
+
+        [Test]
+        public void Constructor_Always_InvokesBaseConstructor()
+        {
+            (var modixContext, var uut) = BuildTestContext();
+
+            uut.ModixContext.ShouldBeSameAs(modixContext);
+        }
+
+        #endregion Constructor() Tests
+
+        #region BeginMaintainTransactionAsync() Tests
+
+        [Test]
+        [NonParallelizable]
+        public async Task BeginMaintainTransactionAsync_MaintainTransactionIsInProgress_WaitsForCompletion()
+        {
+            (_, var uut) = BuildTestContext();
+
+            var existingTransaction = await uut.BeginMaintainTransactionAsync();
+
+            var result = uut.BeginMaintainTransactionAsync();
+
+            result.IsCompleted.ShouldBeFalse();
+
+            existingTransaction.Dispose();
+            (await result).Dispose();
+        }
+
+        [Test]
+        [NonParallelizable]
+        public async Task BeginMaintainTransactionAsync_MaintainTransactionIsNotInProgress_ReturnsImmediately()
+        {
+            (_, var uut) = BuildTestContext();
+
+            var result = uut.BeginMaintainTransactionAsync();
+
+            result.IsCompleted.ShouldBeTrue();
+
+            (await result).Dispose();
+        }
+
+        [Test]
+        [NonParallelizable]
+        public async Task BeginMaintainTransactionAsync_Always_TransactionIsForContextDatabase()
+        {
+            (var modixContext, var uut) = BuildTestContext();
+
+            var database = Substitute.ForPartsOf<DatabaseFacade>(modixContext);
+            modixContext.Database.Returns(database);
+
+            using (var transaction = await uut.BeginMaintainTransactionAsync())
+            { }
+
+            await database.ShouldHaveReceived(1)
+                .BeginTransactionAsync();
+        }
+
+        #endregion BeginMaintainTransactionAsync() Tests
+
+        #region CreateAsync() Tests
+
+        [Test]
+        public async Task CreateAsync_DataIsNull_DoesNotUpdateEmojisAndThrowsException()
+        {
+            (var modixContext, var uut) = BuildTestContext();
+
+            await Should.ThrowAsync<ArgumentNullException>(uut.CreateAsync(null));
+
+            modixContext.Emoji.Select(x => x.Id).ShouldBe(Emoji.Entities.Select(x => x.Id));
```

Comment:
For the sake of clarity and idiot-proffing, you should specify the optional parameter `ignoreOrder: true` for all the `.ShouldBe()` calls like this.

We could also take it a step further an make a `.ShouldBeSetEqual()` extension method, or something similar.